src/jbuf/jbuf.c

```diff
@@ -418,19 +420,18 @@ int jbuf_put(struct jbuf *jb, const struct rtp_header *hdr, void *mem)
 	f->hdr = *hdr;
 	f->mem = mem_ref(mem);
 
-	jb->nf = 1;
-	LIST_FOREACH(&jb->packetl, le)
-	{
-		struct packet *cur_p = le->data;
-		if (!le->next)
-			break;
-
-		struct packet *next_p = le->next->data;
-
-		/* Count not equal timestamp frames (e.g. video) */
-		if (cur_p->hdr.ts != next_p->hdr.ts)
-			++jb->nf;
+	equal = false;
+	if (f->le.prev) {
+		fc = f->le.prev->data;
+		equal = (fc->hdr.ts == f->hdr.ts);
 	}
+	else if (f->le.next) {
+		fc = f->le.next->data;
+		equal = (fc->hdr.ts == f->hdr.ts);
+	}
+
+	if (!equal)
+		++jb->nf;
```

Comment:
1.  1, 1 arrives -> appended and counted
2.  2, 1 arrives -> appended and not counted
3.  4, 2 arrives -> appended and counted
4.  3, 2 arrives -> inserted and not counted, because equal timestamp with `le->next`